packages/template-retail-react-app/app/components/shopper-agent/index.jsx

```diff
@@ -77,50 +103,192 @@ const validateCommerceAgentSettings = (commerceAgent) => {
 }
 
 /**
- * Checks if the shopper agent is enabled and running on client side.
+ * Checks if the shopper agent only runs when explicitly enabled
+ * and when executing in a browser environment.
  *
- * @param {string} enabled - String representation of enabled state
- * @returns {boolean} True if enabled is 'true' and running on client
+ * @param {string} enabled - String representation of enabled state ('true' or 'false')
+ * @returns {boolean} True if enabled is 'true' and running on client, false otherwise
  */
 const isEnabled = (enabled) => {
     return enabled === 'true' && onClient
 }
 
 /**
  * Internal component that renders the embedded messaging window.
- * Handles event listeners for embedded messaging lifecycle events.
+ * This component handles the lifecycle of the Salesforce Embedded Messaging service,
+ * including script loading, initialization, event handling, and cleanup.
+ *
+ * Key responsibilities:
+ * - Loads the embedded messaging script using useScript hook
+ * - Initializes the MIAW service using useMiaw hook
+ * - Sets up prechat fields with current locale, currency, and user context
+ * - Manages event listeners for messaging lifecycle events
+ * - Handles z-index management for maximized chat windows
+ * - Cleans up resources on unmount
  *
  * @param {Object} props - Component props
- * @param {Object} props.commerceAgentConfiguration - Commerce agent configuration
- * @param {string} props.locale - The locale for the embedded messaging script
- * @param {string} props.basketId - The basket ID for the embedded messaging script
- * @returns {null} This component doesn't render any visible UI
+ * @param {Object} props.commerceAgentConfiguration - Commerce agent configuration object
+ * @param {string} props.commerceAgentConfiguration.embeddedServiceName - Name of the embedded service
+ * @param {string} props.commerceAgentConfiguration.embeddedServiceEndpoint - Service endpoint URL
+ * @param {string} props.commerceAgentConfiguration.scriptSourceUrl - Script source URL
+ * @param {string} props.commerceAgentConfiguration.scrt2Url - SCRT2 URL
+ * @param {string} props.commerceAgentConfiguration.salesforceOrgId - Salesforce org ID
```

Comment:
Re-iterating this [comment](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/62c0ef60e5ebb5ac63f0c0f209f6ac59157d64e8/packages/template-retail-react-app/app/components/shopper-agent/index.jsx#L135) from the previous [PR](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/3258) going into this feature branch. Better to address before this merges into `develop` to avoid piling tech debt.